File: apps/api/internal/domain/medicine.go
package domain

import "time"

// Expiration statuses, one per LED color: green, yellow and red.
const (
	ExpirationStatusGood   = "good"
	ExpirationStatusNear   = "near"
	ExpirationStatusExpire = "expire"
)

// Medicine is a registered medicine in the inventory.
type Medicine struct {
	ID          string
	Name        string
	Barcode     string
	BatchNumber *string
	// ExpirationDate is a calendar date; only its year, month and day are meaningful.
	ExpirationDate time.Time
	Quantity       int
	// Location is the tray compartment (1–12), or nil when not placed.
	Location  *int
	CreatedBy *string
	CreatedAt time.Time
	UpdatedAt time.Time
}

File: apps/app/src/types/medicine.ts
export interface Medicine {
  id: string
  name: string
  barcode: string
  batch_number: string | null
  expiration_date: string
  quantity: number
  // Tray compartment (1–12), or null when not placed.
  location: number | null
  created_by: string | null
  created_at: string
  updated_at: string
}

export interface MedicineListResponse {
  data: Medicine[]
  total: number
  limit: number
  offset: number
}

export interface CreateMedicineInput {
  name: string
  barcode: string
  batch_number: string | null
  expiration_date: string
  quantity: number
  location: number | null
}

export interface UpdateMedicineInput {
  name: string
  barcode: string
  batch_number: string | null
  expiration_date: string
  // Always sent: the API clears an omitted location.
  location: number | null
}

export interface ErrorResponse {
  error: string
}
